io-engine/src/lvs/lvs_store.rs

```diff
@@ -880,31 +881,48 @@ impl Lvs {
     pub async fn grow(&self) -> Result<(), LvsError> {
         info!("{self:?}: growing lvs...");
 
-        let cname = self.base_bdev().name().into_cstring();
-        let uri_str = &self.base_bdev().bdev_uri_str().unwrap_or_default();
-        let url = Url::parse(uri_str).map_err(|source| LvsError::InvalidBdev {
+        let disk_bdev = self
+            .base_bdev()
+            .crypto_base_bdev()
+            .map(Bdev::new)
+            .unwrap_or_else(|| self.base_bdev());
+
+        let uri_str = disk_bdev.bdev_uri_str().unwrap_or_default();
+        let url = Url::parse(&uri_str).map_err(|source| LvsError::InvalidBdev {
             source: BdevError::UriParseFailed {
                 source,
                 uri: uri_str.to_string(),
             },
             name: self.name().to_string(),
         })?;
+
         if url.scheme() != "malloc" {
+            let bdev = disk_bdev.name().into_cstring();
             info!(
-                "Attempting to rescan bdev {:?} part of lvs {:?}, uri {:?}",
-                self.base_bdev().name(),
+                "Attempting to rescan bdev {:?} part of lvs {:?}",
+                bdev,
```

Comment:
```suggestion
                "Attempting to rescan bdev {bdev:?} part of lvs {:?}",
```

Also I see we removed the uri, can we not use `uri_str` here?

Reply:
Is it required? We are resizing the bdev right? This message has lvs and bdev name.

Reply:
Only mentioned as you had it before, could be useful

Reply:
Added it back.